core/src/com/unciv/ui/saves/QuickSave.kt

```diff
@@ -0,0 +1,97 @@
+package com.unciv.ui.saves
+
+import com.unciv.Constants
+import com.unciv.MainMenuScreen
+import com.unciv.UncivGame
+import com.unciv.logic.GameInfo
+import com.unciv.ui.crashhandling.launchCrashHandling
+import com.unciv.ui.crashhandling.postCrashHandlingRunnable
+import com.unciv.ui.popup.Popup
+import com.unciv.ui.popup.ToastPopup
+import com.unciv.ui.worldscreen.WorldScreen
+
+
+//todo reduce code duplication
+
+object QuickSave {
+    fun save(gameInfo: GameInfo, screen: WorldScreen) {
+        val gameSaver = UncivGame.Current.gameSaver
+        val toast = ToastPopup("Quicksaving...", screen)
+        launchCrashHandling("SaveGame", runAsDaemon = false) {
+            gameSaver.saveGame(gameInfo, "QuickSave") {
+                postCrashHandlingRunnable {
+                    toast.close()
+                    if (it != null)
+                        ToastPopup("Could not save game!", screen)
+                    else {
+                        ToastPopup("Quicksave successful.", screen)
+                    }
+                }
+            }
+        }
+    }
+
+    fun load(screen: WorldScreen) {
+        val gameSaver = UncivGame.Current.gameSaver
+        val toast = ToastPopup("Quickloading...", screen)
+        launchCrashHandling("LoadGame") {
+            try {
+                val loadedGame = gameSaver.loadGameByName("QuickSave")
+                postCrashHandlingRunnable {
+                    toast.close()
+                    UncivGame.Current.loadGame(loadedGame)
+                    ToastPopup("Quickload successful.", screen)
+                }
+            } catch (ex: Exception) {
+                postCrashHandlingRunnable {
```

Comment:
Toast should be closed on unsuccessful attempt as well, maybe the whole thread should be taken out of try catch

Reply:
This was just _moved_ code, no attempt yet to improve on it.

The whole file is a TODO meant for later, the advantage being to have all save/load _UI_ code in one package and folder, out of WorldScreen, making such improvements easier. And before anyone asks - yea I _know_ maps _aren't_ reusing this for the common usecases but should do so.